holoviews/operation/datashader.py

```diff
@@ -987,16 +987,31 @@ class trimesh_rasterize(aggregate):
     def _precompute(self, element, agg):
         from datashader.utils import mesh
         if element.vdims and getattr(agg, 'column', None) not in element.nodes.vdims:
-            simplices = element.dframe([0, 1, 2, 3])
-            verts = element.nodes.dframe([0, 1])
+            simplex_dims = [0, 1, 2, 3]
+            vert_dims = [0, 1]
         elif element.nodes.vdims:
-            simplices = element.dframe([0, 1, 2])
-            verts = element.nodes.dframe([0, 1, 3])
+            simplex_dims = [0, 1, 2]
+            vert_dims = [0, 1, 3]
+        else:
+            raise ValueError("Cannot shade TriMesh without value dimension.")
+        if set([element.interface.datatype, element.nodes.interface.datatype]) == {'dask'}:
+            dims, node_dims = element.dimensions(), element.nodes.dimensions()
+            simplices = element.data[[dims[sd].name for sd in simplex_dims]]
+            verts = element.nodes.data[[node_dims[vd].name for vd in vert_dims]]
+        else:
```

Comment:
Might be nice to issue a warning if a mixture of pandas and dask dataframes are present to say they are all being converted to pandas.